lib/extract-svg-metadata.js

```diff
@@ -1,23 +1,37 @@
-const createConfig = require('svgo/lib/svgo/config');
-const convertSVG = require('svgo/lib/svgo/svg2js');
-const createSVG = require('svgo/lib/svgo/js2svg');
-const applyPlugins = require('svgo/lib/svgo/plugins');
+const { resolvePluginConfig } = require('svgo/lib/svgo/config');
+const { parseSvg } = require('svgo/lib/parser');
+const { stringifySvg } = require('svgo/lib/stringifier');
+const { invokePlugins } = require('svgo/lib/svgo/plugins');
 const computePathBounds = require('svg-boundings').path;
 
 const metadataPrefix = 'mapbox-';
 
+// Removes `style` attribute. Otherwise, transforms will not apply. 
+// Refer to https://github.com/svg/svgo/blob/a9834efa1603198ae606765292215462ae193d0d/plugins/applyTransforms.js#L55-L57
+const removeStyleAttribute = {
+    name: 'removeStyleAttribute',
+      type: 'perItem',
+  fn: (item) => {
+    if (item.hasAttr('style'))
+        item.removeAttr('style');
+    return item;
+    }
+};
+
 // Transforms `id="mapbox-*"` into `mapbox:metadata="*"` so that the `convertPathData` step
 // resolves transforms.
 const renameMapboxMetadata = {
+    name: 'renameMapboxMetadata',
     type: 'perItem',
     fn: function(item) {
         if (item.hasAttr('id')) {
             const id = item.attr('id');
             if (id.value.startsWith(metadataPrefix)) {
+                let valueSubstr = id.value.substring(metadataPrefix.length);
```

Comment:
In the latest svgo, `removeAttr` has changed. So when we remove attribute `id` (in the next line), we miss the whole `id.value` (it becomes `undefined`), therefore when we try `substring(...)`, we will receive the error `Cannot read property 'substring' of undefined`.
To fix this, we need to preserve the `substring` before `item.removeAttr('id');`.